src/js/game/entity.js

```diff
@@ -82,14 +83,25 @@ export class Entity extends BasicSerializableObject {
     }
 
     /**
-     * Returns a clone of this entity without contents
+     * Returns a clone of this entity
      */
-    duplicateWithoutContents() {
-        const clone = new Entity(this.root);
+    clone() {
+        const staticComp = this.components.StaticMapEntity;
+        const buildingData = getBuildingDataFromCode(staticComp.code);
+
+        const clone = buildingData.metaInstance.createEntity({
+            root: this.root,
+            origin: staticComp.origin,

```

Comment:
You need to clone the vector here too, since otherwise changing it would change the original vector